autorest/preparer.go

```diff
@@ -183,6 +183,16 @@ func WithBaseURL(baseURL string) PrepareDecorator {
 	}
 }
 
+// WithCustomBaseURL returns a PrepareDecorator that replaces brace-enclosed keys within the
+// request base URL (i.e., http.Request.URL) with the corresponding values from the passed map.
+func WithCustomBaseURL(baseURL string, urlParameters map[string]interface{}) PrepareDecorator {
```

Comment:
Nevermind, I retract the map[string]fmt.Stringer suggestion, I just played with it in [go playground](https://play.golang.org) and don't like how that would impact declaring map literals.